packages/ethereum/src/erc20/connector.ts

```diff
@@ -33,12 +38,12 @@ export class OiErc20 extends OiConnector {
 
       const [caipTagFrom, caipTagTo] = await super.tagAndResolve(fromAccount, toAccount);
       
-      coreApp.log.info(`ERC20 transfer ${amount}, ${fromAccount.toString()}, ${toAccount.toString()}`);
+      plugin.log.info(`ERC20 transfer ${amount}, ${fromAccount.toString()}, ${toAccount.toString()}`);
 
       await super.addLog('Transfer', { block: event.blockNumber, caipTagFrom, caipTagTo, amount, event});
 
-      coreApp.log.info(`ERC20 transfer CAIP-Tags from ${caipTagFrom} (${fromAccount.toString()}), to ${caipTagTo} (${fromAccount.toString()})`);
-      coreApp.log.info(`ERC20 transfer looking up addresses: ${await lookupCaipTag(caipTagFrom)} (${caipTagFrom}), ${await lookupCaipTag(caipTagTo)} (${caipTagTo})`);
+      plugin.log.info(`ERC20 transfer CAIP-Tags from ${caipTagFrom} (${fromAccount.toString()}), to ${caipTagTo} (${fromAccount.toString()})`);
+      plugin.log.info(`ERC20 transfer looking up addresses: ${await lookupCaipTag(caipTagFrom)} (${caipTagFrom}), ${await lookupCaipTag(caipTagTo)} (${caipTagTo})`);
     }, 
     20740100 );

```

Comment:
There is a hardcoded FROM-block (this one: https://etherscan.io/block/20740100)
This makes the ethereum-module exlusively usable for Ethereum Mainnet after Sep 13, 2024. I assume this should either be a config option or the FROM-Block shall be detected from the blockchain itself?

Reply:
ContractFactory will get a "getDeploymentBlock" method. This was added (my fault) to test the indexer. 

Your error above could be related to the decimals and a few other minor differences between Standard erc20 and USDC

Reply:
IMO this should be done still in this branch, to fix main asap. AFAIK on the other hand main is already broken this way - so we could also fix it independently.

Clear preference to fix it immediately though!

Reply:
Created #12